File: examples/chill/trmm.f
      SUBROUTINE trmm(N,A,B,C)
      INTEGER N
      REAL*8  A(N,N), B(N,N), C(N,N)
      
      INTEGER I,J,K

      DO 10,K=1,N
         DO 20,I=K,N
            DO 30, J=K,N
               C(I,J) = C(I,J)+A(I,K)*B(K,J)
 30         CONTINUE
 20      CONTINUE
 10   CONTINUE

      END

File: examples/chill/testcases/trmm.c

#define N       1024


void trmm(double A[N][N], double B[N][N], double C[N][N]) {
    int i, j, k;
    for(k = 0; k < N; k++) {
        for(i = 0; i < N; i++) {
            for(j = 0; j < N; j++) {
                C[j][i] = C[j][i] + A[k][i] * B[j][k];
            }
        }
    }
}


